lib/api/controllers/baseController.js

```diff
@@ -92,6 +99,248 @@ class BaseController {
 
     return Boolean(flagValue);
   }
+
+  /**
+   * Extracts an array parameter from the request input
+   *
+   * @param {Request} request - Request object
+   * @param {String} paramPath - Path of the parameter to extract (eg: 'body.foo')
+   * @param {?String[]} defaultValue
+   *
+   * @throws
+   * @returns {String[]}
+   */
+  getArrayParam (request, paramPath, defaultValue = null) {
+    const arrayParam = _.get(request.input, paramPath, defaultValue);
+
+    if (arrayParam === null) {
+      errorsManager.throw('missing_param', paramPath);
+    }
+
+    if (! Array.isArray(arrayParam)) {
+      errorsManager.throw('invalid_param_type', paramPath, arrayParam, 'Array');
+    }
+
+    return arrayParam;
+  }
+
+  /**
+   * Extracts a string parameter from the request input
+   *
+   * @param {Request} request - Request object
+   * @param {String} paramPath - Path of the parameter to extract (eg: 'body.foo')
+   * @param {?String[]} defaultValue
+   *
+   * @throws
+   * @returns {String[]}
+   */
+  getStringParam (request, paramPath, defaultValue = null) {
+    const stringParam = _.get(request.input, paramPath, defaultValue);
+
+    if (stringParam === null) {
+      errorsManager.throw('missing_param', paramPath);
+    }
+
+    if (typeof stringParam !== 'string') {
+      errorsManager.throw('invalid_param_type', paramPath, stringParam, 'String');
+    }
+
+    return stringParam;
+  }
+
+  /**
+   * Extracts an object parameter from the request input
+   *
+   * @param {Request} request - Request object
+   * @param {String} paramPath - Path of the parameter to extract (eg: 'body.foo')
+   * @param {?String[]} defaultValue
+   *
+   * @throws
+   * @returns {String[]}
+   */
+  getObjectParam (request, paramPath, defaultValue = null) {
+    const objectParam = _.get(request.input, paramPath) || defaultValue;
+
+    if (objectParam === null) {
+      errorsManager.throw('missing_param', paramPath);
+    }
+
+    if (! _.isPlainObject(objectParam)) {
+      errorsManager.throw('invalid_param_type', paramPath, objectParam, 'Object');
+    }
+
+    return objectParam;
+  }
+
+  /**
+   * Extracts a number parameter from the request input
+   *
+   * @param {Request} request - Request object
+   * @param {String} paramPath - Path of the parameter to extract (eg: 'body.foo')
+   * @param {?String[]} defaultValue
+   *
+   * @throws
+   * @returns {Number}
+   */
+  getNumberParam (request, paramPath, defaultValue = null) {
+    const rawNumberParam = _.get(request.input, paramPath, defaultValue);
+
+    if (rawNumberParam === null) {
+      errorsManager.throw('missing_param', paramPath);
+    }
+
+    const numberParam = parseInt(rawNumberParam);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add the base to this "parseInt" call. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2427)